### ISSUE TITLE
Report validation metrics for all `n_forward_steps`

### ISSUE THREAD

**Issue by oliverwm1** [ai2cm/ace#390]
Currently, our validation aggregator only reports metrics based on the first forward step. This makes sense when training on only one forward step. But if optimizing over multiple steps, it is difficult to assess performance on later steps (the loss is the only thing reported for the later steps).

I propose following solution:

In `validate_one_epoch` ([link](https://github.com/ai2cm/full-model/blob/b924e64e60ee67389813bee6ce8cef469ccd3cff/fme/fme/ace/train/train.py#L351)) initialize a list of `OneStepAggregator` of length `n_forward_steps`. Then when recording batches [here](https://github.com/ai2cm/full-model/blob/b924e64e60ee67389813bee6ce8cef469ccd3cff/fme/fme/ace/train/train.py#L376), for each aggregator, slice the target/gen data to the appropriate pair of time steps. E.g. `(0, 1)`, `(1,2)`, ... , `(n_forward_steps-1, n_forward_steps)`.

Then call `.get_logs` on each of these with appropriate labels (`val/step_{n}`?) and merge the dicts before returning. For backwards compatibility with existing logs we could have a special case for step 0 where we just label with `val`. Also, the loss returned by `run_on_batch` is a special case because it doesn't correspond to a particular step, so should also just log it under the `val` label.

**Comment by mcgibbon** [comment]
Not sure what it was like when we made this issue, but with 20 forward steps and the number of validation metrics we use, I'm guessing we don't want this in the way it's stated. But maybe we can get some channel-mean reduced metrics for each step? Or specific things like the precipitation power spectrum bias?

**Comment by oliverwm1** [comment]
Yes I agree the naive implementation described above would end up computing too many metrics. I think the per-step loss is sufficient for reporting.
